pkg/odo/cli/component/deploy.go

```diff
@@ -135,9 +134,12 @@ func (do *DeployOptions) Run() (err error) {
 	}
 
 	manifestURL := metadata.Manifest
+	if manifestURL == "" {
+		return errors.New("Unable to deploy as alpha.deployment-manifest is not defined in devfile.yaml")
+	}
```

Comment:
For another PR, need to verify that this URL is valid

Reply:
There is a `util.ValidateURL` function already that we can use

Reply:
amazing

Reply:
Hmmm, doesn't seem to fail when an invalid URL is passed!!!

Reply:
OK, only does basic validation of URL